transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollTcpInfo.java

```diff
@@ -61,133 +61,133 @@
  */
 public final class EpollTcpInfo {
 
-    final int[] info = new int[32];
+    final long[] info = new long[32];
 
     public int state() {
-        return info[0] & 0xFF;
+        return (int) info[0];
```

Comment:
What about introducing a new int safe cast utility method? 

From openjdk 8..

```java
/**
     * Returns the value of the {@code long} argument;
     * throwing an exception if the value overflows an {@code int}.
     *
     * @param value the long value
     * @return the argument as an int
     * @throws ArithmeticException if the {@code argument} overflows an int
     * @since 1.8
     */
    public static int toIntExact(long value) {
        if ((int)value != value) {
            throw new ArithmeticException("integer overflow");
        }
        return (int)value;
    }
```